go/src/koding/kites/kloud/metadata/cloudinit_test.go

```diff
@@ -0,0 +1,130 @@
+package metadata_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"koding/kites/kloud/metadata"
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+var update = flag.Bool("update-golden", false, "Update golden files.")
+
+var testdata = metadata.NewCloudInit(&metadata.CloudConfig{
+	Username:  "johndoe",
+	Metadata:  "metadata",
+	Userdata:  "userdata",
+	KiteKey:   "kitekey",
+	Provision: "provision",
+})
+
+func parseCloudInit(file string) (metadata.CloudInit, error) {
+	p, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	return metadata.ParseCloudInit(p)
+}
+
+func TestCloudInit(t *testing.T) {
+	cases := map[string]struct {
+		golden   string
+		conflict []string
+	}{
+		"testdata/ssh_authorized_keys.yml": {
+			golden:   "testdata/ssh_authorized_keys.yml.golden",
+			conflict: nil,
+		},
+		"testdata/users.yml": {
+			golden:   "testdata/users.yml.golden",
+			conflict: nil,
+		},
+		"testdata/write_files.yml": {
+			golden:   "testdata/write_files.yml.golden",
+			conflict: nil,
+		},
+		"testdata/conflict_output.yml": {
+			conflict: []string{"output", "all"},
+		},
+		"testdata/conflict_users.yml": {
+			conflict: []string{"users", "1", "name"},
+		},
+		"testdata/conflict_write_files.yml": {
+			conflict: []string{"write_files", "1", "path"},
+		},
+	}
+
+	for yml, cas := range cases {
+		t.Run(filepath.Base(yml), func(t *testing.T) {
+			if *update {
+				ci, err := parseCloudInit(yml)
```

Comment:
just a personal preference, i would write the code for updating in a diff function since it is not directly related with the tests + it could decrease the function body line count